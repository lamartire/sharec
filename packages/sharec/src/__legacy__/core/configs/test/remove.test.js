const { fixture } = require('testUtils')
const { vol } = require('memfs')
const { removeConfig } = require('../remove')

describe('processors > configs >', () => {
  // TODO: remove snapshots tests
  const eslint04 = fixture('eslint/json/eslintrc_04.json', 'json')
  const eslint05 = fixture('eslint/json/eslintrc_05.json', 'json')
  const yamlEslint04 = fixture('eslint/yaml/eslintrc_04.yml')
  const yamlEslint05 = fixture('eslint/yaml/eslintrc_05.yml')

  beforeEach(() => {
    vol.reset()
  })

  describe('removeConfig', () => {
    it('should remove configs from exist JSON file', async () => {
      expect.assertions(1)

      const dir = {
        '/target/.eslintrc': JSON.stringify(eslint04),
        '/configuration-package/.eslintrc': JSON.stringify(eslint05),
      }

      vol.fromJSON(dir)

      await removeConfig({
        configsPath: '/configuration-package',
        targetPath: '/target',
        filePath: '.eslintrc',
      })

      expect(
        JSON.parse(vol.readFileSync('/target/.eslintrc', 'utf8')),
      ).toMatchSnapshot()
    })

    it('should remove configs from exist YAML file', async () => {
      expect.assertions(1)

      const dir = {
        '/target/.eslintrc.yaml': yamlEslint04,
        '/configuration-package/.eslintrc.yaml': yamlEslint05,
      }

      vol.fromJSON(dir)

      await removeConfig({
        configsPath: '/configuration-package',
        targetPath: '/target',
        filePath: '.eslintrc.yaml',
      })

      expect(
        vol.readFileSync('/target/.eslintrc.yaml', 'utf8'),
      ).toMatchSnapshot()
    })

    it('should delete exist JSON config file if target file was not changed', async () => {
      expect.assertions(1)

      const dir = {
        '/target/.eslintrc': JSON.stringify(eslint04),
        '/configuration-package/.eslintrc': JSON.stringify(eslint04),
      }

      vol.fromJSON(dir)

      await removeConfig({
        configsPath: '/configuration-package',
        targetPath: '/target',
        filePath: '.eslintrc',
      })

      expect(vol.readdirSync('/target')).not.toContain('.eslintrc')
    })
  })
})
