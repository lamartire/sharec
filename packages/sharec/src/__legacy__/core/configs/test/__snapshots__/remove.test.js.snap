// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`processors > configs > removeConfig should remove configs from exist JSON file 1`] = `
Object {
  "env": Object {
    "node": false,
  },
  "extends": "standard",
  "globals": Object {
    "baz": true,
  },
  "rules": Object {
    "max-len": Array [
      "warn",
      80,
      2,
    ],
    "sort-imports": 0,
  },
}
`;

exports[`processors > configs > removeConfig should remove configs from exist YAML file 1`] = `
"env:
  node: false
extends: standard
globals:
  baz: true
rules:
  max-len:
    - warn
    - 80
    - 2
  sort-imports: 0
"
`;
